File: freak-fights-fan/shared/features/dictionaryItems/queries/GetAllMyDictionaryItems.ts
import { SortOrder } from "../../../abstractions/SortOrder";
import { IPagedQuery } from "../../../abstractions/IPagedQuery";
import { ISortedQuery } from "../../../abstractions/ISortedQuery";

export namespace GetAllMyDictionaryItems {
    export interface Query extends IPagedQuery, ISortedQuery {
        page: number;
        pageSize: number;
        sortColumn: string;
        sortOrder: SortOrder;
        searchTerm: string;

        dictionaryId: number;
    }
}

File: FreakFightsFan.Shared/Features/DictionaryItems/Queries/GetAllMyDictionaryItems.cs
using FluentValidation;
using FreakFightsFan.Shared.Abstractions;
using FreakFightsFan.Shared.Features.DictionaryItems.Responses;
using MediatR;

namespace FreakFightsFan.Shared.Features.DictionaryItems.Queries;

public static class GetAllMyDictionaryItems
{
    public class Query : IRequest<PagedList<MyDictionaryItemDto>>, IPagedQuery, ISortedQuery
    {
        public int Page { get; set; }
        public int PageSize { get; set; }
        public string SortColumn { get; set; }
        public SortOrder SortOrder { get; set; }
        public string SearchTerm { get; set; }

        public int DictionaryId { get; set; }
    }

    public class Validator : AbstractValidator<Query>
    {

    }
}